model/common/src/icon4py/model/common/test_utils/serialbox_utils.py

```diff
@@ -453,8 +454,8 @@ def construct_icon_grid(self, on_gpu: bool) -> icon.IconGrid:
         )
         c2e2c = self.c2e2c()
         e2c2e = self.e2c2e()
-        c2e2c0 = xp.column_stack(((range(c2e2c.shape[0])), c2e2c))
-        e2c2e0 = xp.column_stack(((range(e2c2e.shape[0])), e2c2e))
+        c2e2c0 = xp.column_stack((xp.asarray(range(c2e2c.shape[0])), c2e2c))
```

Comment:
you have this `on_gpu` flag as an argument in this function. I don't actually now whether it is always set correctly and we might trade it for a backen argument as well, but you could do the conditional import of the `array_ns` based on that.

Reply:
I have decided to use numpy for reading these connectivities from serialized data and used `array_ns` and `on_gpu` to decide between `cp` or `np` in `_get_offset_provider` of `BaseGrid` class. It is enough to make the driver run on gpu or
cpu depending on the backend. (I think all those remaining fields can also be read in as numpy arrays. I am not sure about the strategy, so I did not change the remaining part in `serialbox_utils.py`).